libraries/AP_Proximity/AP_Proximity_LightWareSF40C.cpp

```diff
@@ -166,280 +197,288 @@ void AP_Proximity_LightWareSF40C::init_sectors()
+                            _distance[last_sector] = last_distance_cm * 0.01f;
+                            _distance_valid[last_sector] = true;
+                        } else {
+                            _distance_valid[last_sector] = false;
+                        }
+                        // update boundary used for avoidance
+                        update_boundary_for_sector(last_sector, true);
+                    }
+                    // initialise for new sector
+                    last_sector = sector;
+                    last_distance_cm = INT16_MAX;
+                }
+                // use shortest valid distance
+                if ((dist_cm < last_distance_cm) && (dist_cm >= dist_min_cm) && (dist_cm <= dist_max_cm)) {
+                    last_distance_cm = dist_cm;
+                    last_angle_deg = angle_deg;
+                }
             }
-            break;
         }
-
-        default:
-            break;
+        break;
     }
-
-    // mark request as cleared
-    if (success) {
-        _last_request_type = RequestType_None;
+    case MessageID::MOTOR_STATE:
+        if (msg.payload_recv == 1) {
+            sensor_state.motor_state = msg.payload[0];
+        }
+        break;
+    case MessageID::OUTPUT_RATE:
+        if (msg.payload_recv == 1) {
+            sensor_state.output_rate = msg.payload[0];
+        }
+        break;
+
+    // unsupported messages
+    case MessageID::PRODUCT_NAME:
+    case MessageID::HARDWARE_VERSION:
+    case MessageID::FIRMWARE_VERSION:
+    case MessageID::SERIAL_NUMBER:
+    case MessageID::TEXT_MESSAGE:
+    case MessageID::USER_DATA:
+    case MessageID::SAVE_PARAMETERS:
+    case MessageID::STAGE_FIRMWARE:
+    case MessageID::COMMIT_FIRMWARE:
+    case MessageID::INCOMING_VOLTAGE:
+    case MessageID::LASER_FIRING:
+    case MessageID::TEMPERATURE:
+    case MessageID::BAUD_RATE:
+    case MessageID::DISTANCE:
+    case MessageID::MOTOR_VOLTAGE:
+    case MessageID::FORWARD_OFFSET:
+    case MessageID::REVOLUTIONS:
+    case MessageID::ALARM_STATE:
+    case MessageID::ALARM1:
+    case MessageID::ALARM2:
+    case MessageID::ALARM3:
+    case MessageID::ALARM4:
+    case MessageID::ALARM5:
+    case MessageID::ALARM6:
+    case MessageID::ALARM7:
+        break;
     }
+}
 
-    return success;
+// convert buffer to uint32, uint16
+uint32_t AP_Proximity_LightWareSF40C::buff_to_uint32(uint8_t b0, uint8_t b1, uint8_t b2, uint8_t b3) const
+{
+    uint32_t leval = (uint32_t)b0 | (uint32_t)b1 << 8 | (uint32_t)b2 << 16 | (uint32_t)b3 << 24;
+    return le32toh(leval);
 }
 
-// clear buffers ahead of processing next message
-void AP_Proximity_LightWareSF40C::clear_buffers()
+uint16_t AP_Proximity_LightWareSF40C::buff_to_uint16(uint8_t b0, uint8_t b1) const
 {
-    element_len[0] = 0;
-    element_len[1] = 0;
-    element_num = 0;
-    memset(element_buf, 0, sizeof(element_buf));
+    uint16_t leval = (uint16_t)b0 | (uint16_t)b1 << 8;
+    return le16toh(leval);
```

Comment:
Same. Actually you use UINT16_VALUE macro in one place in code - which is correct.

Reply:
ok, removed the le16toh